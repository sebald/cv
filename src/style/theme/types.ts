import * as Styled from 'styled-components';
import theme from '.';
import palette from './palette';

// Theme
// ---------------
export type Theme = typeof theme;

export type ThemedStyledProps<P = {}> = Styled.ThemedStyledProps<P, Theme>;

export type ThemedInterpolation = Styled.Interpolation<
  Styled.ThemeProps<Theme>
>;

export type ThemedInterpolationFunction<P = {}> = Styled.InterpolationFunction<
  Styled.ThemeProps<Theme> & P
>;

export type StyledMixin<P = {}> = ThemedInterpolationFunction<P>;

// Theme Props
// ---------------
export type TypographyStyles =
  | 'display'
  | 'heading'
  | 'subheading'
  | 'caption'
  | 'body';
export type Color = keyof typeof palette;
