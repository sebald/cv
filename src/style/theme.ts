export const theme = {
  baseline: 12,
  htmlFontSize: 16,
};
