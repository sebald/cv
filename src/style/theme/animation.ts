import { keyframes } from '../styled-components';

export const fadeInUp = keyframes`
  from {
    opacity: 0;
    transform: translateY(30px);
  }

  to {
    opacity: 1;
    transform: translateY(0);
  }
`;
