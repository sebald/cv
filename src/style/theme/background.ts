import palette from './palette';

export default {
  body: palette.white,
};
