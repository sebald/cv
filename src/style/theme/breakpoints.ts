export default {
  small: 768,
  medium: 1024,
  large: 1280,
};
