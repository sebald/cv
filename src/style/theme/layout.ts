export default {
  sidebar: 350,
  content: 800,
};
