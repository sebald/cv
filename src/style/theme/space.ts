export default {
  baseline: 16,
};
