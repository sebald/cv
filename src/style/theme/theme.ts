import breakpoints from './breakpoints';
import layout from './layout';
import palette from './palette';
import space from './space';
import typography from './typography';

const theme = {
  breakpoints,
  layout,
  palette,
  space,
  typography,
};

export default theme;
