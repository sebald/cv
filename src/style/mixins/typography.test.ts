import theme from '../theme';
import {
  fontWeight,
  getHtmlFontSize,
  textAlign,
  textTransform,
  typography,
} from './typography';

test('get html font size', () => {
  expect(getHtmlFontSize({ theme })).toMatchSnapshot();
});

test('typography', () => {
  expect(typography('body')({ theme })).toMatchSnapshot();
  expect(typography('caption')({ theme })).toMatchSnapshot();
});

test('fontWeight', () => {
  expect(fontWeight({ fontWeight: 200 } as any));
  expect(fontWeight({} as any));
});

test('textAlign', () => {
  expect(textAlign({ textAlign: 'center' } as any));
  expect(textAlign({} as any));
});

test('textTransform', () => {
  expect(textTransform({ textTransform: 'uppercase' } as any));
  expect(textTransform({} as any));
});
