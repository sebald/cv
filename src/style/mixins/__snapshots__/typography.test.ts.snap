// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get html font size 1`] = `16`;

exports[`typographyStyle 1`] = `
Object {
  "color": "#424242",
  "fontFamily": "'Raleway', sans-serif;",
  "fontSize": "1rem",
  "fontWeight": 300,
  "lineHeight": "2rem",
  "marginBottom": "1.375rem",
  "marginTop": 0,
  "paddingTop": "0.625rem",
}
`;

exports[`typographyStyle 2`] = `
Object {
  "color": "#9e9e9e",
  "fontFamily": "'Montserrat', sans-serif;",
  "fontSize": "1rem",
  "fontWeight": 300,
  "lineHeight": "2rem",
  "marginBottom": "1.375rem",
  "marginTop": 0,
  "paddingTop": "0.625rem",
}
`;
